starlark/value.go

```diff
@@ -698,6 +698,7 @@ func (fn *Function) Globals() StringDict { return fn.module.makeGlobalDict() }
 
 func (fn *Function) Position() syntax.Position { return fn.funcode.Pos }
 func (fn *Function) NumParams() int            { return fn.funcode.NumParams }
+func (fn *Function) NumParamDefaults() int     { return len(fn.defaults) }
```

Comment:
This isn't necessary. Let's keep it simple: the caller can call ParamDefault for each of the NumParams.

Reply:
The length of `NumParams()` and `fn.defaults` may be different though right? `fn.defaults` only has values for parameters that have a default (and keyword-only parameters).

Reply:
Exactly, the representation omits the longest prefix that would otherwise contain all "mandatory" elements. But we shouldn't expose that in the API.

Reply:
Gotcha, done.